File: server/src/main/java/com/example/bmi/BmiService.java
package com.example.bmi;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class BmiService {
    /*
       Author: Azamat Afzalov
       Matriculation number: 1492864
       Date: 05.11.2023
    */
    private final BmiRepository bmiRepository;
    @Autowired
    public BmiService(BmiRepository bmiRepository) {
        this.bmiRepository = bmiRepository;
    }
    public Bmi create(Bmi body) {
        return this.bmiRepository.save(body);
    }
    public List<Bmi> getAll() {
        return this.bmiRepository.findAll();
    }

    public Double calculateBmi(Double height, Double weight) {
        return (weight) / Math.pow(height / 100, 2);
    }

    public Optional<Bmi> getById(Long id) {
        return this.bmiRepository.findById(id);
    }
}

File: bmi-client/src/app/bmi.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import { environment } from '../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class BmiService {
  /*
    Author: Azamat Afzalov
    Matriculation number: 1492864
    Date: 05.11.2023
  */
  constructor(
    private http: HttpClient,
  ) { }

  getAll() {
    return this.http.get('http://20.172.68.200:8080/api/v1/bmi');
  }

  submitForm(body: any) {
    return this.http.post('http://20.172.68.200:8080/api/v1/bmi', body)
  }
}
